### PR TITLE
Add node and pod snapshots

### PR DESCRIPTION
**Pull request by tbak**

### CODE REVIEW

**Comment by tbak** [comment]
@joshi-keyur Merging early as it is blocking my testing. Please, leave your comments and I will address them in a separate PR.

File: pod/snapshot.go [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+	// Pods with the primary resource pool being this one
+	Primary map[string]*k8sCore.Pod
+}
+
+type Options struct {
+	SupportGPUs        bool
+	PastYoungThreshold func(pod *k8sCore.Pod, now time.Time) bool
+}
+
+func NewEmpty() *Snapshot {
```
Comment (joshi-keyur) [inline pod/snapshot.go:37]: Maybe the naming can be improved so it is clearer when referenced in the context of the `pod` package .

File: pod/snapshot.go [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+		ScheduledByName:   map[string]*k8sCore.Pod{},
+		FinishedByName:    map[string]*k8sCore.Pod{},
+		Metadata:          map[string]*Metadata{},
+		Primary:           map[string]*k8sCore.Pod{},
+	}
+}
+
+// Returns Snapshot of pods associated with the given resource pool and the list of the remaining pods.
+// This factory does not filter out running pods on node resources not belonging to this resource pool
+// (expected if a pod is associated with many resource pools). This additional filtering step can be done by calling
```
Comment (joshi-keyur) [inline pod/snapshot.go:51]: Is this being done for future proofing (ref: pod associated with many resource pools)?

Comment (tbak) [reply pod/snapshot.go:51]: No, it is done on purpose. This version does not require a caller to have node list available.

File: pod/snapshot.go (hunk 1 above)
Comment (tbak) [reply pod/snapshot.go:37]: Like `NewEmptySnapshot`? Could you give an example?

**Pull request merged**